src/lib/MAVLink/MAVLink.cpp

```diff
@@ -92,9 +95,10 @@ void convert_mavlink_to_crsf_telem(uint8_t *CRSFinBuffer, uint8_t count, Handset
                 crsffm = {0};
                 ap_flight_mode_name4(crsffm.p.flight_mode, ap_vehicle_from_mavtype(heartbeat.type), heartbeat.custom_mode);
                 // if we have a good flight mode, and we're armed, suffix the flight mode with a * - see Ardupilot's AP_CRSF_Telem::calc_flight_mode()
-                if (strlen(crsffm.p.flight_mode) == 4 && (heartbeat.base_mode & MAV_MODE_FLAG_SAFETY_ARMED)) {
-                    crsffm.p.flight_mode[4] = '*';
-                    crsffm.p.flight_mode[5] = '\0';
+                size_t len = strlen(crsffm.p.flight_mode);
```

Comment:
For safety we should really use `strnlen`.

Reply:
Totally fair - added.